### PR TITLE
verify-libs-and-licenses 5a / 5b

### PR DESCRIPTION
**Pull request by vieiro**
5a. db.sql.visualeditor/external/javacc-3.2-license.txt
	contains a license body which does not match that in
	nbbuild/licenses/BSD: mismatch around:
	'any nuclear facility' vs. 'ILITY OF SUCH DAMAGE'...

	javacc is used at compile time only, not at runtime.
	Added commetns to javacc-3.2-license.txt clarifying this.
	There's a "non nuclear facility" clause in the BSD license,
	I think this is ok. If not we can always upgrade to more
	recent javacc versions, where the clause is not present.

	5b. libs.bytelist/external/bytelist-0.1-license.txt
	has a non-header line in the header block: " JRuby, and ...

	We were depending on the wrong binary. Updated binary
	dependency to org.jruby.util. Maven reported this as
	covered by the MIT license, but it's really licensed
	under GPL3/LGPL3/EPL1.0. We choose EPL1.0
	Added a comment to bytelist-1.0.2-license.txt with
	a link to bytelist's repo clarifying license.

### CODE REVIEW

File: db.sql.visualeditor/external/javacc-3.2-license.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@ Description: Parser generator for SQL parser.
 Origin: https://javacc.dev.java.net/files/documents/17/686/javacc-3.2.zip
 OSR: 3106
 License: BSD
+Comment: JavaCC is only used to compile a parser. javacc binaries are not included in the module's binaries.
```
Comment (matthiasblaesing) [inline db.sql.visualeditor/external/javacc-3.2-license.txt]: There is a header to express this. Please add this header:

```
Type: compile-time
```

That way the ant modules know how to handle this (used in VerifyLibsAndLicenses, CreateDependencies)

File: libs.bytelist/external/bytelist-1.0.2-license.txt [hunk 2]
```diff
@@ -0,0 +1,95 @@
+Name: Bytelist
+Version: 1.0.2
+Description: The ByteList library from JRuby.
+License: EPL-v10
+Origin: https://github.com/jruby/bytelist
+Comment: This library was wrongly reported in Maven as released under the MIT license. It's licensed under the GPL 3, LGPL 3 or EPL 1.0 see: https://github.com/jruby/bytelist/commit/9a0d9de4d21f6fc4342f946166d43cf3bf8b798a for the upgrade.
```
Comment (matthiasblaesing) [inline libs.bytelist/external/bytelist-1.0.2-license.txt]: This assessment is not correct. You are are referring to git master and that is indeed triple licensed as described. That is not true for version 1.0.2:

https://github.com/jruby/bytelist/blob/4e41c255c4d58328e3a422fc79651a0464b55600/src/org/jruby/util/ByteList.java

This is CPL 1.0 + GPL 2.0 + LGPL 2.1

I would do this:
- move to 1.0.15
- document, that the POM info is wrong (https://github.com/jruby/bytelist/commit/9a0d9de4d21f6fc4342f946166d43cf3bf8b798a#diff-600376dffeb79835ede4a0b285078036)
- add info, that the sources file next to the binary file clearly states the correct EPL/GPL/LGPL triple license
- bump specification version (0.30.0) as API got bigger

I ran the two jars through japi-compliance-checker and the result was really positive. The attached zip contains the HTML report and bytelist gained new methods, but kept the old ones.

[compat_report.html.zip](https://github.com/apache/incubator-netbeans/files/1549356/compat_report.html.zip)

**Comment by vieiro** [comment]
Hi Matthias,

Thanks for the thorough review! You're my hero!

The "Type: compile-time" flag only works if license is "GPL-2-CP" in VerifyLibsAndLicenses.java, shall we update the ant task to include BSD licenses as well?

**Comment by jlahoda** [comment]
Antonio, "Type: compile-time" should be supported for all licenses (if not, that's a bug I think). But for GPL-2-CP, both "Type: compile-time" and a "Comment" are required.

**Comment by vieiro** [comment]
We'll have to modify VerifyLibsAndLicenses.java then, as  "compile-time" is only checked for GPL licenses (see [VerifyLibsAndLicenses.java:line 306](https://github.com/apache/incubator-netbeans/blob/9.0-alpha-rc2/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/VerifyLibsAndLicenses.java))

Would you like me to add it to this PR or want a specific PR?
Also, is this pseudo-code correct?

```
    if (contains "GPL") {
        "Type: compile-time" is required.
        if (contains "GPL-2-CP") {
            "Comment" is required.
        }
    } 
    if ("Type: compile-time") {
        ignore rest of license checking
    }
```

**Comment by jlahoda** [comment]
The (intent of the) current checks is that "Type: compile-time" is required for GPL-2-CP (aongside with a "Comment"). Because we are not allowed to have a hard dependency on GPL code. It is OK to have a hard dependency on BSD, but it is also OK to have a "Type: compile-time" BSD dependency.

**Comment by vieiro** [comment]
Then I think VerifyLibsAndLicenses is perfect.
We then have to decide if the "non nuclear" clause in javacc license is aceptable or not. If not we can upgrade the version, I think.

**Comment by matthiasblaesing** [comment]
One thing that is missing is, that the license text matches the text in `nbbuild/licenses/*`. I suggest to: add a license template `nbbuild/licenses/BSD-javacc` with the license body of `db.sql.visualeditor/external/javacc-3.2-license.txt`. And update the `javacc-3.2-license.txt` file to a license named `BSD-javacc` (the performance module also depends on javacc and should be updated accordingly).

I found this by running:
```
ant verify-all-libs-and-licenses
```

**Comment by vieiro** [comment]
Thanks, Matthias.
I've upgraded the "performance" module that also uses javacc 3.2 to reuse this license. One line less in the verifylibsandlicenses file.

**Comment by matthiasblaesing** [comment]
Great. I think you should merge this.

**Pull request merged**